src/test/java/org/elasticsearch/bwcompat/OldIndexBackwardsCompatibilityTests.java

```diff
@@ -70,6 +84,76 @@ public static void initIndexes() throws Exception {
         Collections.sort(indexes);
     }
     
+    @AfterClass
+    public static void tearDownStatics() {
+        indexes = null;
+        indicesDir = null;
+    }
+    
+    @Override
+    public Settings nodeSettings(int ord) {
+        return ImmutableSettings.builder()
+            .put(Node.HTTP_ENABLED, true) // for _upgrade
+            .put(MergePolicyModule.MERGE_POLICY_TYPE_KEY, NoMergePolicyProvider.class) // disable merging so no segments will be upgraded
+            .build();
+    }
+    
+    void setupDataLoadingNodes() throws Exception {
+        Path dataDir = newTempDirPath(LifecycleScope.SUITE);
+        indicesDir = dataDir.resolve(internalCluster().getClusterName()).resolve("nodes/0/indices/");
```

Comment:
Sure, then just pick the first node and the first path.

On Thu, Mar 26, 2015 at 6:12 PM, Ryan Ernst <notifications@github.com>
wrote:

>> +    public static void tearDownStatics() {
>> +        indexes = null;
>> +        indicesDir = null;
>> +    }
>> +    
>> +    @Override
>> +    public Settings nodeSettings(int ord) {
>> +        return ImmutableSettings.builder()
>> +            .put(Node.HTTP_ENABLED, true) // for _upgrade
>> +            .put(MergePolicyModule.MERGE_POLICY_TYPE_KEY, NoMergePolicyProvider.class) // disable merging so no segments will be upgraded
>> +            .build();
>> +    }
>> +    
>> +    void setupDataLoadingNodes() throws Exception {
>> +        Path dataDir = newTempDirPath(LifecycleScope.SUITE);
>> +        indicesDir = dataDir.resolve(internalCluster().getClusterName()).resolve("nodes/0/indices/");
> Regarding the randomization, I'm trying very hard *not* make this test highly randomized.  I don't think the purpose of this test should be to test random configurations, etc.  I would like debugging to be very clear and not surprising.  Having a single node path (or 2 sets of paths in the future) is much more direct and debuggable (I will add a logging line for the path(s)). 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/elastic/elasticsearch/pull/10247/files#r27233851